client/src/components/Sider/AdminSider.tsx

```diff
@@ -10,12 +9,30 @@ import { Course } from 'services/models';
 import { getAdminMenuItems, getCourseManagementMenuItems } from './data/menuItems';
 const { Sider } = Layout;
 
+type MenuItem = Required<MenuProps>['items'][number];
+
 type Props = { session: Session; courses: Course[]; activeCourse?: Course | null };
 
 enum LocalStorage {
   IsSiderCollapsed = 'isSiderCollapsed',
 }
 
+function getItem(
+  label: React.ReactNode,
+  key: React.Key,
+  icon?: React.ReactNode,
+  children?: MenuItem[],
+  type?: 'group',
+): MenuItem {
+  return {
+    key,
+    icon,
+    children,
+    label,
+    type,
+  } as MenuItem;
```

Comment:
We don't need typecast here :)
```suggestion
  };
```